apps/emqx_bridge_kafka/src/emqx_bridge_kafka_action_info.erl

```diff
@@ -23,15 +23,23 @@ connector_type_name() -> kafka_producer.
 
 schema_module() -> emqx_bridge_kafka.
 
-action_to_bridge_v1_fixup(Config) ->
-    emqx_utils_maps:rename(<<"parameters">>, <<"kafka">>, Config).
-
-bridge_v1_to_action_fixup(Config0) ->
-    Config = emqx_utils_maps:rename(<<"kafka">>, <<"parameters">>, Config0),
-    maps:with(producer_action_field_keys(), Config).
+connector_action_config_to_bridge_v1_config(ConnectorConfig, ActionConfig) ->
+    BridgeV1Config1 = maps:remove(<<"connector">>, ActionConfig),
+    BridgeV1Config2 = emqx_utils_maps:deep_merge(ConnectorConfig, BridgeV1Config1),
+    emqx_utils_maps:rename(<<"parameters">>, <<"kafka">>, BridgeV1Config2).
+
+bridge_v1_config_to_action_config(BridgeV1Conf, ConnectorName) ->
+    BridgeV1Conf,
```

Comment:
```suggestion
```

Reply:
I think the compiler should complain about things like this...